backend/services/carddeck/handler/carddeck.go

```diff
@@ -217,13 +217,68 @@ func (e *CardDeck) CreateDeck(ctx context.Context, req *pb.CreateDeckRequest, rs
 	return nil
 }
 
+func (e *CardDeck) CopyDeck(ctx context.Context, req *pb.CopyDeckRequest, rsp *pb.DeckResponse) error {
+	logger.Infof("Received CardDeck.CopyDeck request: %v", req)
+	if err := e.checkUserDeckAccess(ctx, req.UserID, req.DeckID); err != nil {
+		return err
+	}
+	if err := e.checkUserRoleAccess(ctx, req.UserID, req.TargetGroupID, pbCollaboration.GroupRole_WRITE); err != nil {
+		return err
+	}
+	var (
+		deckType model.DeckType
+		err      error
+	)
+	if req.DeckType == nil {
+		deckType = model.PrivateDeckType
+	} else {
+		err, deckType = converter.MigrateProtoDeckTypeToModelDeckType(*req.DeckType)
```

Comment:
swap this if-statement

```suggestion
	if req.DeckType != nil {
		deckType, err = converter.MigrateProtoDeckTypeToModelDeckType(*req.DeckType)
	} else {
		deckType = model.PrivateDeckType
```

Reply:
implemented in 2b00ace